### PR TITLE
Organization Refactor (WIP)

### PR DESCRIPTION
**Pull request by bonedaddy**

### CODE REVIEW

File: models/organization.go [hunk 1]
```diff
@@ -186,6 +186,28 @@ func (om *OrgManager) GetTotalStorageUsed(name string) (uint64, error) {
...
+	usr, err := NewUserManager(om.DB).FindByUserName(username)
+	if err != nil {
+		return nil, err
+	}
+	// prevent returning upload results for users that are not part of the organization
+	// this is to prevent exploiting users from examining other peoples data, however
+	// we still need to implement an auth check in the caller.
+	if usr.Organization != orgName {
+		return nil, errors.New("user does not belong to organization")
+	}
```
Comment (xiegeo) [inline models/organization.go]: since we are checking for orgName matching, I don't think the first `om.FindByName(orgName)` is needed

Comment (bonedaddy) [reply models/organization.go]: If we do that, and an empty organization name is given, then it's possible for the caller to retrieve uploads of all non organization users

Comment (bonedaddy) [reply models/organization.go]: Granted we will prevent this with a check in the caller within Temporal's API, I thinkt this is a good check to have an avoid accidentally disclosing data.

Comment (xiegeo) [reply models/organization.go]: ok

**Pull request merged**